rest/server_context.go

```diff
@@ -42,14 +41,38 @@ const DefaultStatsLogFrequencySecs = 60
 // This struct is accessed from HTTP handlers running on multiple goroutines, so it needs to
 // be thread-safe.
 type ServerContext struct {
-	config            *ServerConfig
-	databases_        map[string]*db.DatabaseContext
-	lock              sync.RWMutex
-	statsContext      *statsContext
-	HTTPClient        *http.Client
-	replicator        *base.Replicator
-	cpuPprofFileMutex sync.Mutex // Protect cpuPprofFile from concurrent Start and Stop CPU profiling requests
-	cpuPprofFile      *os.File   // An open file descriptor holds the reference during CPU profiling
+	config           *StartupConfig
+	persistentConfig bool
+	// bucketDbConfigs is a map of bucket name to DatabaseConfig
+	bucketDbConfigs map[string]*DatabaseConfig
```

Comment:
How is bucketDbConfigs going to be used?  Given that, need to consider future compatibility with databases that target collections or sets of collections instead of buckets.

Reply:
Forgot I moved the usage of this into #5057. I'll remove from this PR, but CollectionName->DatabaseConfig would be the equivalent once collections are introduced.